src/host/exe/CConsoleHandoff.cpp

```diff
@@ -0,0 +1,63 @@
+#include "precomp.h"
+
+#include "CConsoleHandoff.h"
+
+#include "srvinit.h"
+
+// Routine Description:
+// - Helper to duplicate a handle to ourselves so we can keep holding onto it
+//   after the caller frees the original one.
+// Arguments:
+// - in - Handle to duplicate
+// - out - Where to place the duplicated value
+// Return Value:
+// - S_OK or Win32 error from `::DuplicateHandle`
+HRESULT _duplicateHandle(const HANDLE in, HANDLE& out)
+{
+    RETURN_IF_WIN32_BOOL_FALSE(DuplicateHandle(GetCurrentProcess(), in, GetCurrentProcess(), &out, 0, FALSE, DUPLICATE_SAME_ACCESS));
+    return S_OK;
+}
+
+// Routine Description:
+// - Takes the incoming information from COM and and prepares a console hosting session in this process.
+// Arguments:
+// - server - Console driver server handle
+// - inputEvent - Event already established that we signal when new input data is available in case the driver is waiting on us
+// - msg - Portable attach message containing just enough descriptor payload to get us started in servicing it
+HRESULT CConsoleHandoff::EstablishHandoff(HANDLE server,
+                                          HANDLE inputEvent,
+                                          PCCONSOLE_PORTABLE_ATTACH_MSG msg)
+try
+{
+    // Fill the descriptor portion of a fresh api message with the received data.
+    // The descriptor portion is the "received" packet from the last ask of the driver.
+    // The other portions are unnecessary as they track the other buffer state, error codes,
+    // and the return portion of the api message.
+    // We will re-retrieve the connect information (title, window state, etc.) when the
+    // new console session begins servicing this.
+    CONSOLE_API_MSG apiMsg;

```

Comment:
may want to zero initialize this-

```suggestion
    CONSOLE_API_MSG apiMsg{};
```

Reply:
Just to be safe. We don't want random memory!